src/db/migrations/20-add-consumable-views.ts

```diff
@@ -0,0 +1,128 @@
+
+import { Knex } from 'knex';
+
+import { Table } from '../db';
+
+async function getForeignKeys(knex: Knex): Promise<{
+  table_name: string,
+  column_name: string,
+  foreign_table_name: string,
+  foreign_column_name: string
+}[]> {
+  const result = await knex.raw(
+    ` 
+    SELECT
+        tc.table_name, 
+        kcu.column_name, 
+        ccu.table_name AS foreign_table_name,
+        ccu.column_name AS foreign_column_name 
+    FROM 
+        information_schema.table_constraints AS tc 
+        JOIN information_schema.key_column_usage AS kcu
+          ON tc.constraint_name = kcu.constraint_name
+          AND tc.table_schema = kcu.table_schema
+        JOIN information_schema.constraint_column_usage AS ccu
+          ON ccu.constraint_name = tc.constraint_name
+          AND ccu.table_schema = tc.table_schema
+    WHERE tc.constraint_type = 'FOREIGN KEY';   
+    `
+  )
+
+  return result.rows
+}
+
+enum oldTables {
+  platforms = 'platforms',
+  nfts = 'nfts',
+  erc721Transfers = 'erc721Transfers',
+  artists = 'artists',
+  artistProfiles = 'artistProfiles',
+  processedTracks = 'processedTracks',
+  processors = 'processors',
+  metaFactories = 'metaFactories',
+  nftFactories = 'nftFactories',
+  nfts_processedTracks = 'nfts_processedTracks',
+  nftProcessErrors = 'nftProcessErrors',
+  ipfsPins = 'ipfsPins',
+}
+
+function tableNameToViewName(tableName: string): string {
+  return `consumable${tableName[0].toUpperCase() + tableName.substring(1)}`
+}
+
+export const up = async (knex: Knex) => {
+
+  // rename tables with rawPrefix
+  for (const key of Object.keys(oldTables)){
+    const oldName = oldTables[key as oldTables];
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    const newName = Table[key as Table];
+    await knex.schema.renameTable(oldName, newName);
+  }
+
+  // specify conditions to add to each view
+  const conditions: {
+    [table in Table]?: string
+  } = {
+    [Table.processedTracks]: `
+    "lossyArtworkIPFSHash" is not null 
+    and "lossyAudioIPFSHash" is not null`,
+  }
```

Comment:
we should also actually ensure that both the audio and artwork are pinned, so join both cids with the pin table and ensure they have an entry there